src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/RazorSemanticTokensLegend.cs

```diff
@@ -2,177 +2,70 @@
-            "json - string",
-            "json - text",
-            "label name",
-            "local name",
-            "method name",
-            "module name",
-            "namespace name",
-            "operator - overloaded",
-            "parameter name", // 50
-            "property name",
-            "preprocessor keyword",
-            "preprocessor text",
-            "punctuation",
-            "record class name",
-            "record struct name",
-            "regex - alternation",
-            "regex - anchor",
-            "regex - character class",
-            "regex - comment", // 60
-            "regex - grouping",
-            "regex - other escape",
-            "regex - quantifier",
-            "regex - self escaped character",
-            "regex - text",
-            "string - escape character",
-            "struct name",
-            "text",
-            "type parameter name",
-            "string - verbatim", // 70
-            "whitespace",
-            "xml doc comment - attribute name",
-            "xml doc comment - attribute quotes",
-            "xml doc comment - attribute value",
-            "xml doc comment - cdata section",
-            "xml doc comment - comment",
-            "xml doc comment - delimiter",
-            "xml doc comment - entity reference",
-            "xml doc comment - name",
-            "xml doc comment - processing instruction", // 80
-            "xml doc comment - text",
-            "xml literal - attribute name",
-            "xml literal - attribute quotes",
-            "xml literal - attribute value",
-            "xml literal - cdata section",
-            "xml literal - comment",
-            "xml literal - delimiter",
-            "xml literal - embedded expression",
-            "xml literal - entity reference",
-            "xml literal - name", // 90
-            "xml literal - processing instruction",
-            "xml literal - text",
-            RazorTagHelperElementType,
-            RazorTagHelperAttributeType,
-            RazorTransitionType,
-            RazorDirectiveColonType,
-            RazorDirectiveAttributeType,
-            RazorDirectiveType,
-            RazorCommentType, // 100
-            RazorCommentTransitionType,
-            RazorCommentStarType,
-            MarkupTagDelimiterType,
-            MarkupElementType,
-            MarkupOperatorType,
-            MarkupAttributeType,
-            MarkupAttributeQuoteType,
-            MarkupTextLiteralType,
-            MarkupCommentPunctuationType,
-            MarkupCommentType, // 110
-            MarkupAttributeValueType,
-            RazorComponentElementType,
-            RazorComponentAttributeType,
-        };
+        // C# types + Razor types
+        public static readonly string[] TokenTypes = RazorSemanticTokensAccessor.RoslynTokenTypes.Concat(
+            typeof(RazorSemanticTokensLegend).GetFields(BindingFlags.NonPublic | BindingFlags.Static).Where(
```

Comment:
We don't get any guarantees about order from `GetFields`. That shouldn't super matter, except that while the Tests still rely on raw numbers a runtime change may break us.